bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommandResult.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.etherrain.internal.api;
+
+/**
+ * The {@link EtherRainCommandResult} is the response packet for Command Result
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+public enum EtherRainCommandResult {
+    RESULT_OK("OK"),
+    RESULT_INTERRUPTED_RAIN("RN"),
+    RESULT_INTERUPPTED_SHORT("SH"),
+    RESULT_INCOMPLETE("NC");
+
+    protected String result;
+
+    EtherRainCommandResult(String result) {
+        this.result = result;
+    }
+
+    public static EtherRainCommandResult fromString(String text) {
+        for (EtherRainCommandResult b : EtherRainCommandResult.values()) {
```

Comment:
If you don't care about the exception (or just handle) you could consider using the existing `valueOf `method:

```java
EtherRainCommandResult.valueOf(text.toUpperCase());
```

And maybe handling the `IllegalArgumentException` which is thrown if there is no such value. And if text can also be `null` you might want to handle that as well. But if its not Nullable (better use the annotations to guarantee that) I would prefer the normal `valueOf`